### ISSUE TITLE
fmt build failure

### ISSUE THREAD

**Issue by DougRogers** [microsoft/vcpkg#16702]
Windows x64/  VS 2019 Community

**To Reproduce**
vcpkg install fmt --triplet x64-windows
```
C:\Users\drogers\Documents\Visual Studio 2010\Projects\vcpkg>vcpkg install fmt --triplet x64-windows
Computing installation plan...
The following packages will be built and installed:
    fmt[core]:x64-windows
Detecting compiler hash for triplet x64-windows...
Starting package 1/1: fmt:x64-windows
Building package fmt[core]:x64-windows...
Could not locate cached archive: C:\Users\drogers\AppData\Local\vcpkg\archives\af\af8b97833c16c51347a08eee5219e633acaa8911.zip
-- Using cached C:/Users/drogers/Documents/Visual Studio 2010/Projects/vcpkg/downloads/fmtlib-fmt-cd4af11efc9c622896a3e4cb599fa28668ca3d05.tar.gz
-- Cleaning sources at C:/Users/drogers/Documents/Visual Studio 2010/Projects/vcpkg/buildtrees/fmt/src/8668ca3d05-7f40cec0ee.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/Users/drogers/Documents/Visual Studio 2010/Projects/vcpkg/downloads/fmtlib-fmt-cd4af11efc9c622896a3e4cb599fa28668ca3d05.tar.gz
-- Applying patch fix-warning4189.patch
-- Using source at C:/Users/drogers/Documents/Visual Studio 2010/Projects/vcpkg/buildtrees/fmt/src/8668ca3d05-7f40cec0ee.clean
-- Configuring x64-windows
-- Building x64-windows-dbg
-- Building x64-windows-rel
-- Installing: C:/Users/drogers/Documents/Visual Studio 2010/Projects/vcpkg/packages/fmt_x64-windows/share/fmt/copyright
-- Using msys root at C:/Users/drogers/Documents/Visual Studio 2010/Projects/vcpkg/downloads/tools/msys2/c809757c94447846
-- Fixing pkgconfig file: C:/Users/drogers/Documents/Visual Studio 2010/Projects/vcpkg/packages/fmt_x64-windows/lib/pkgconfig/fmt.pc
-- CHECK_LIB_fmt_RELEASE:CHECK_LIB_fmt_RELEASE-NOTFOUND
CMake Error at scripts/cmake/vcpkg_fixup_pkgconfig.cmake:232 (message):
  Library "fmt" was not found! If it is a system library use the
  SYSTEM_LIBRARIES parameter for the vcpkg_fixup_pkgconfig call! Otherwise,
  correct the *.pc file
Call Stack (most recent call first):
  scripts/cmake/vcpkg_fixup_pkgconfig.cmake:297 (vcpkg_fixup_pkgconfig_check_files)
  ports/fmt/portfile.cmake:44 (vcpkg_fixup_pkgconfig)
  scripts/ports.cmake:79 (include)


Error: Building package fmt:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: fmt:x64-windows
  Vcpkg version: 2020.06.15-nohash

Additionally, attach any relevant sections from the log files above.
```

**Comment by JonLiu1993** [comment]
@DougRogers ,Thanks for posting this issue,This is because the vcpkg path contains capital letters. Are there any uppercase characters in the path of your vcpkg inventory?

**Comment by JackBoosY** [comment]
```
CMake Error at scripts/cmake/vcpkg_fixup_pkgconfig.cmake:232 (message):
  Library "fmt" was not found! If it is a system library use the
  SYSTEM_LIBRARIES parameter for the vcpkg_fixup_pkgconfig call! Otherwise,
  correct the *.pc file
```

Please re-install `fmt`.

Thanks.

**Comment by SZenglein** [comment]
> Please update vcpkg (using `git pull`) and re-install `fmt`.
> 
> Thanks.

Why update vcpkg when there has not been a commit in the last four days? Anyway, I also experience this issue.

**Comment by DougRogers** [comment]
I updated vckpg using .\bootstrap-vcpkg.bat


Same error:
```
C:\Users\drogers\Documents\Visual Studio 2010\Projects\vcpkg>.\bootstrap-vcpkg.bat

Building vcpkg.exe ...

  pch.cpp
  archives.cpp
  checks.cpp
  chrono.cpp
  cofffilereader.cpp
  downloads.cpp
  enums.cpp
  files.cpp
  hash.cpp
  json.cpp
  machinetype.cpp
  parse.cpp
  strings.cpp
  stringview.cpp
  system.cpp
  system.print.cpp
  system.process.cpp
  unicode.cpp
  binarycaching.cpp
  binaryparagraph.cpp
  build.cpp
  buildenvironment.cpp
  cmakevars.cpp
  commands.autocomplete.cpp
  commands.buildexternal.cpp
  commands.cache.cpp
  commands.ci.cpp
  commands.ciclean.cpp
  commands.contact.cpp
  commands.cpp
  commands.create.cpp
  commands.dependinfo.cpp
  commands.edit.cpp
  commands.env.cpp
  commands.fetch.cpp
  commands.format-manifest.cpp
  commands.hash.cpp
  commands.info.cpp
  commands.integrate.cpp
  commands.list.cpp
  commands.owns.cpp
  commands.porthistory.cpp
  commands.portsdiff.cpp
  commands.search.cpp
  commands.setinstalled.cpp
  commands.upgrade.cpp
  commands.upload-metrics.cpp
  commands.version.cpp
  commands.xvsinstances.cpp
  configuration.cpp
  dependencies.cpp
  export.cpp
  export.chocolatey.cpp
  export.ifw.cpp
  export.prefab.cpp
  globalstate.cpp
  help.cpp
  input.cpp
  install.cpp
  platform-expression.cpp
  metrics.cpp
  packagespec.cpp
  paragraphs.cpp
  portfileprovider.cpp
  postbuildlint.buildtype.cpp
  postbuildlint.cpp
  registries.cpp
  remove.cpp
  sourceparagraph.cpp
  statusparagraph.cpp
  statusparagraphs.cpp
  tools.cpp
  triplet.cpp
  update.cpp
  userconfig.cpp
  vcpkgcmdarguments.cpp
  vcpkglib.cpp
  vcpkgpaths.cpp
  versiont.cpp
  visualstudio.cpp
  vcpkg.cpp
  Generating code
  Previous IPDB not found, fall back to full compilation.
  All 20134 functions were compiled because no usable IPDB/IOBJ from previous compilation was found.
  Finished generating code
  vcpkg.vcxproj -> C:\Users\drogers\Documents\Visual Studio 2010\Projects\vcpkg\toolsrc\windows-bootstrap\msbuild.x86.r
  elease\vcpkg.exe

Building vcpkg.exe... done.

Telemetry
---------
vcpkg collects usage data in order to help us improve your experience.
The data collected by Microsoft is anonymous.
You can opt-out of telemetry by re-running the bootstrap-vcpkg script with -disableMetrics,
passing --disable-metrics to vcpkg on the command line,
or by setting the VCPKG_DISABLE_METRICS environment variable.

Read more about vcpkg telemetry at docs/about/privacy.md

C:\Users\drogers\Documents\Visual Studio 2010\Projects\vcpkg>vcpkg install fmt --triplet x64-windows
Computing installation plan...
The following packages will be built and installed:
    fmt[core]:x64-windows
Detecting compiler hash for triplet x64-windows...
Starting package 1/1: fmt:x64-windows
Building package fmt[core]:x64-windows...
Could not locate cached archive: C:\Users\drogers\AppData\Local\vcpkg\archives\af\af8b97833c16c51347a08eee5219e633acaa8911.zip
-- Using cached C:/Users/drogers/Documents/Visual Studio 2010/Projects/vcpkg/downloads/fmtlib-fmt-cd4af11efc9c622896a3e4cb599fa28668ca3d05.tar.gz
-- Cleaning sources at C:/Users/drogers/Documents/Visual Studio 2010/Projects/vcpkg/buildtrees/fmt/src/8668ca3d05-7f40cec0ee.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/Users/drogers/Documents/Visual Studio 2010/Projects/vcpkg/downloads/fmtlib-fmt-cd4af11efc9c622896a3e4cb599fa28668ca3d05.tar.gz
-- Applying patch fix-warning4189.patch
-- Using source at C:/Users/drogers/Documents/Visual Studio 2010/Projects/vcpkg/buildtrees/fmt/src/8668ca3d05-7f40cec0ee.clean
-- Configuring x64-windows
-- Building x64-windows-dbg
-- Building x64-windows-rel
-- Installing: C:/Users/drogers/Documents/Visual Studio 2010/Projects/vcpkg/packages/fmt_x64-windows/share/fmt/copyright
-- Using msys root at C:/Users/drogers/Documents/Visual Studio 2010/Projects/vcpkg/downloads/tools/msys2/c809757c94447846
-- Fixing pkgconfig file: C:/Users/drogers/Documents/Visual Studio 2010/Projects/vcpkg/packages/fmt_x64-windows/lib/pkgconfig/fmt.pc
-- CHECK_LIB_fmt_RELEASE:CHECK_LIB_fmt_RELEASE-NOTFOUND
CMake Error at scripts/cmake/vcpkg_fixup_pkgconfig.cmake:232 (message):
  Library "fmt" was not found! If it is a system library use the
  SYSTEM_LIBRARIES parameter for the vcpkg_fixup_pkgconfig call! Otherwise,
  correct the *.pc file
Call Stack (most recent call first):
  scripts/cmake/vcpkg_fixup_pkgconfig.cmake:297 (vcpkg_fixup_pkgconfig_check_files)
  ports/fmt/portfile.cmake:44 (vcpkg_fixup_pkgconfig)
  scripts/ports.cmake:79 (include)


Error: Building package fmt:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: fmt:x64-windows
  Vcpkg version: 2020.06.15-nohash

Additionally, attach any relevant sections from the log files above.


```

**Comment by sfhacker** [comment]
Are we still having issues with fmt?

**Comment by JackBoosY** [comment]
```
-- Using source at C:/Users/drogers/Documents/Visual Studio 2010/Projects/vcpkg/buildtrees/fmt/src/8668ca3d05-7f40cec0ee.clean
```
fmt source path should be `2b3b374f0b-8ea71ee8ea.clean` in latest. I think your repo is not correct.
BTW, vcpkg support Visual Studio 2015 or later, and please DO NOT use the path contains empty space.

**Comment by DougRogers** [comment]
If the repo is not correct after running _bootstrap-vcpkg.bat_m then it does update properly.

The directory name is legacy.  I am using VC 2019  as I indicated above.  This was the default directory where projects were installed for VS 2010 and that is where they stayed as I upgraded VS.  

I can move vcpkg to a path without spaces, but it is surprising and disappointing that this is required.  I did not have any problems with vcpkg before this.

After downloading from github directly, rebuilding _vcpkg.exe_ and moving to a non-spaces directory, _fmt_ downloaded and built properly.
